hydroflow/src/scheduled/mod.rs

```diff
@@ -113,6 +116,23 @@ impl Hydroflow {
         }
     }
 
+    /**
+     * Run the dataflow graph, blocking until the closure returns true.
+     */
+    pub fn run_while<F>(&mut self, f: F) -> Result<(), RecvError>
+    where
+        F: Fn() -> bool,
+    {
+        while f() {
+            // Do any current work.
+            self.tick();
+
+            // Block and wait for an external event.
+            self.ready_queue.push_back(self.event_queue_recv.recv()?);
+        }
+        Ok(())
+    }
```

Comment:
This method seems a little silly, maybe `self.ready_queue.push_back(self.event_queue_recv.recv()?);` should be put in its own pub method like `wait_for_events()` or something and then you can take this out as
```
while !*(*done).borrow() {
  df.tick();
  df.wait_for_events();
}
```
Probably the whole `tick()`, `run()`, `enqueue_jobs()` methods need a bit of tidying up

Reply:
Done